src/utils/vscodeUtils.ts

```diff
@@ -101,3 +104,25 @@ export function getNodeEditorLabel(node: IAzureNode): string {
 function isAccountTreeItem(treeItem: IAzureTreeItem): boolean {
     return (treeItem instanceof MongoAccountTreeItem) || (treeItem instanceof DocDBAccountTreeItemBase);
 }
+
+export function getDocumentTreeItemLabel(document: IMongoDocument | RetrievedDocument): string {
+    for (let field of documentLabelFields) {
+        if (document.hasOwnProperty(field)) {
+            let value = document[field];
+            if (value) { //ignore if false-y value: null, undefined, "".
```

Comment:
This will also ignore zero, which I don't think would be desirable.

Reply:
Good catch. 

Looking at the other [falsy values](https://developer.mozilla.org/en-US/docs/Glossary/Falsy), I think we'll be covered after acconting for 0.

Reply:
In this case I'd suggest the following:

I'd suggest this:
(note: you're effectively checking that it's not an object with your value.toString().startsWith("[object ")) below.  I'm not sure this idea behavior when the property to display is an object or array, but it seems reasonable to skip it and try something else.  But typeof is a lot better than the toString.startsWith()

// If doesn't have value, or is null, ObjectId or other object type, try a different property
if (typeof value !== 'undefined' && typeof value !== 'object') {
    return String(value);
}

I think that handles it all, right?

Reply:
Yeah. It relies on null having type object (slightly obfuscated). If that's well known, then I'll go ahead with it.

Reply:
Unfortunately, typeof null = "object" in JS.  Well-known "bug" that can't be fixed now.